addons/voice/org.openhab.voice.pollytts/src/main/java/org/openhab/voice/pollytts/internal/PollyTTSService.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+    /**
+     * obtain audio stream from cache or Amazon Polly service and return it to play the audio
+     */
+    @Override
+    public AudioStream synthesize(String inText, Voice voice, AudioFormat requestedFormat) throws TTSException {
+        logger.debug("Synthesize '{}' in format {}", inText, requestedFormat);
+        logger.debug("voice UID: '{}' voice label: '{}' voice Locale: {}", voice.getUID(), voice.getLabel(),
+                voice.getLocale());
+
+        // Validate arguments
+        // trim text
+        String text = inText.trim();
+        if (text == null || text.isEmpty()) {
+            throw new TTSException("The passed text is null or empty");
+        }
+        if (!voices.contains(voice)) {
+            throw new TTSException("The passed voice is unsupported");
+        }
+        boolean isAudioFormatSupported = audioFormats.stream()
+                .filter(audioFormat -> audioFormat.isCompatible(requestedFormat)).findAny().isPresent();
+
+        if (!isAudioFormatSupported) {
+            throw new TTSException("The passed AudioFormat is unsupported");
+        }
+
+        // now create the input stream for given text, locale, format. There is
+        // only a default voice
+        try {
+            File cacheAudioFile = pollyTTSImpl.getTextToSpeechAsFile(text, voice.getLabel(),
+                    getApiAudioFormat(requestedFormat));
+            if (cacheAudioFile == null) {
+                throw new TTSException("Could not read from PollyTTS service");
+            }
+            logger.debug("Audio Stream for '{}' in format {}", text, requestedFormat);
+            AudioStream audioStream = new PollyTTSAudioStream(cacheAudioFile, requestedFormat);
+            return audioStream;
+        } catch (AudioException ex) {
+            throw new TTSException("Could not create AudioStream: " + ex.getMessage(), ex);
+        } catch (IOException ex) {
+            throw new TTSException("Could not read from PollyTTS service: " + ex.getMessage(), ex);
+        }
+    }
+
+    private Set<Voice> initVoices() {
+        // @formatter:off
+        return pollyTTSImpl.getAvailableLocales().stream()
+            .flatMap(locale ->
+                pollyTTSImpl.getAvailableVoices(locale).stream()
+                    .map(label -> new PollyTTSVoice(locale, label)))
+            .collect(toSet());
+        // @formatter:on
+    }
+
+    private Set<AudioFormat> initAudioFormats() {
+        // @formatter:off
+        return pollyTTSImpl.getAvailableAudioFormats().stream()
+                .map(this::getAudioFormat)
+                .collect(toSet());
+        // @formatter:on
+    }
+
+    private AudioFormat getAudioFormat(String apiFormat) {
+        if ("MP3".equals(apiFormat)) {
+            // use by default: MP3, 22khz_16bit_mono with bitrate 64 kbps
+            return new AudioFormat(CONTAINER_NONE, CODEC_MP3, null, 16, 64000, 22050L);
+        } else if ("OGG".equals(apiFormat)) {
```

Comment:
```suggestion
        } else if (CONTAINER_OGG.equals(apiFormat)) {
```

Or define an own `static final String` for it.